smack-experimental/src/test/java/org/jivesoftware/smackx/spoiler/SpoilerTest.java

```diff
@@ -0,0 +1,90 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.spoiler;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNull;
+import static org.custommonkey.xmlunit.XMLAssert.assertXMLEqual;
+
+import org.jivesoftware.smack.DummyConnection;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.test.util.SmackTestSuite;
+import org.jivesoftware.smack.test.util.TestUtils;
+import org.jivesoftware.smackx.spoiler.element.SpoilerElement;
+import org.jivesoftware.smackx.spoiler.provider.SpoilerProvider;
+
+import org.junit.Test;
+import org.xmlpull.v1.XmlPullParser;
+
+public class SpoilerTest extends SmackTestSuite {
+
+    private static final XMPPConnection connection = new DummyConnection();
+
+    @Test
+    public void emptySpoilerTest() throws Exception {
+        final String xml = "<spoiler xmlns='urn:xmpp:spoiler:0'/>";
+
+        SpoilerManager m = SpoilerManager.getInstanceFor(connection);
+        SpoilerElement empty = m.createSpoiler();
+
+        assertNull(empty.getHint());
+        assertNull(empty.getLanguage());
+
+        assertXMLEqual(xml, empty.toXML().toString());
+
+        XmlPullParser parser = TestUtils.getParser(xml);
+        SpoilerElement parsed = new SpoilerProvider().parse(parser);
```

Comment:
The common Smack idiom is to have a public field with an instance of a provider, usually for testing purposes, e.g. `SpoilerProvider.INSTANCE`.